### DISCUSSION TITLE
Does useSWR is good for SEO.

### DISCUSSION THREAD

**Discussion by BluetyIt** [vercel/swr#1433] [Q&A] [answered]
Hello to all,
I have few small questions I didn't find an answer in your documentation. The major question is about useSWR and SEO: does useSWR is adapted/suits SEO requirements?
Here is **use-case**:
1. Web page has few date sources loaded by API. (Example, I have "product" page, and I have 3 API paths to get data: ProductInfo, productComments, productRatings)
2. Web page loads the first bunch of information server-side(Example, ProductInfo)
3. Web page gets ProductInfo and got renderer, but still need to get productComments and productRatings via API.
4. Web page has 2 empty blocks (divs) reserved for productComments productRatings
5. Web page loads productComments productRatings using useSWR (imagine to load all data we need to spend 3 seconds)
6. Web page completely loaded with all data.

**Questions:**
1. Does Google, Bing etc, will get the data about productComments productRatings, loaded by useSWR, and will it "reference" this data?
2. Can we useSWR to create web pages with good SEO? I like to load all data by useSWR, and not, for example with getServerSideProps. I know this is kind of strange in the context of nextjs (for better SEO they recommend to use getStaticProps), but I like to have some kind of loaders on each page with huge amount of data. For example, in my context this page can be "listofProducts" - data can be up to 5mb to load.

Using: useSWR, nextjs, vercel.

Thanks for your help!

**Comment by shuding** [comment, marked as answer]
Yes if you use SWR with Next.js’ `getServerSideProps` or `getStaticProps`, the data will be pre-rendered in the markup and it’s perfect for SEO. You can check the corresponding section in the docs.

If you’re not using those static generation strategies and only have client side rendering/data fetching, only _some_ search engines might be able to parse it.
